src/main/java/org/commcare/formplayer/application/MenuController.java

```diff
@@ -165,6 +167,7 @@ public EntityDetailListResponse getDetails(@RequestBody SessionNavigationBean se
     public BaseResponseBean navigateSessionWithAuth(@RequestBody SessionNavigationBean sessionNavigationBean,
             @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken,
             HttpServletRequest request) throws Exception {
+        String locale = sessionNavigationBean.getLocale();
```

Comment:
How does HQ get to know of current locale to send here ?

Reply:
Honestly not too sure! I just assumed all navigation request included the locale in the request body which gets set when initializing the sessionNavigationBean. Happy to be told otherwise but it's been accurately giving me the correct locale so far.

Reply:
did you mean HQ was already sending the current locale in the request body without us having to make a corresponding HQ change here ?

Reply:
yes I believe so!